server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -303,26 +343,67 @@ public void putSynonymsSet(String synonymSetId, SynonymRule[] synonymsSet, Actio
                         : UpdateSynonymsResultStatus.UPDATED;
 
                     reloadAnalyzers(synonymSetId, false, bulkInsertResponseListener, updateSynonymsResultStatus);
-                }));
+                })
+            );
         }));
     }
 
-    public void putSynonymRule(String synonymsSetId, SynonymRule synonymRule, ActionListener<SynonymsReloadResult> listener) {
-        checkSynonymSetExists(synonymsSetId, listener.delegateFailure((l1, obj) -> {
-            try {
-                IndexRequest indexRequest = createSynonymRuleIndexRequest(synonymsSetId, synonymRule).setRefreshPolicy(
-                    WriteRequest.RefreshPolicy.IMMEDIATE
-                );
-                client.index(indexRequest, l1.delegateFailure((l2, indexResponse) -> {
-                    UpdateSynonymsResultStatus updateStatus = indexResponse.status() == RestStatus.CREATED
-                        ? UpdateSynonymsResultStatus.CREATED
-                        : UpdateSynonymsResultStatus.UPDATED;
+    // Open for testing adding more synonyms set than the limit allows for
```

Comment:
Let's add this function in the test directly since it's not used anywhere else?

Reply:
I refactored the code so the bulk insert uses this function, and exposed that for testing - it's actually used from the `SynonymsManagementAPIService` itself.

Reply:
ah ok missed that thanks